dockerfiles/aws/golang/Dockerfile.build

```diff
@@ -0,0 +1,46 @@
+ARG BASE_IMAGE
+FROM ${BASE_IMAGE}
+
+# Define Go version from VERSION build argument
+ARG VERSION
+ENV GO_VERSION=${VERSION}
+ARG TARGETARCH=amd64
+
+#######################################
+# Install required utilities
+RUN yum install -y shadow-utils wget ca-certificates tar zip
+
+# Install gosu for running commands as another user
+ENV GOSU_VERSION 1.14
+# https://github.com/tianon/gosu/releases/tag/1.14
+# key https://keys.openpgp.org/search?q=tianon%40debian.org
+RUN curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/${GOSU_VERSION}/gosu-amd64" \
+    && chmod +x /usr/local/bin/gosu
+
+# Download and install Go
+RUN wget -q "https://go.dev/dl/go${GO_VERSION}.linux-${TARGETARCH}.tar.gz" -O go.tar.gz && \
```

Comment:
If the base image is `amazon/aws-lambda-go:1.2024.10.04.19`, then do we need to install Go separately?